src/parse.rs

```diff
@@ -156,22 +159,36 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
                 Fingerprint {
                     title: dir_name.to_owned(),
                     hash,
+                    flavor: flavor.to_owned(),
                 }
             }
         })
         // Note: we filter out cases where hashing has failed.
         .filter(|f| f.hash.is_some())
         .collect();
 
+    // Get the other flavors, we want saved as well.
+    let other_flavors_fingerprints: Vec<_> = fingerprint_collection
+        .clone()
+        .into_iter()
+        .filter(|f| &f.flavor != flavor)
+        .collect();
+
     // Update our in memory collection and save to disk.
     fingerprint_collection.drain(..);
-    fingerprint_collection.extend(new_fingerprints);
+    fingerprint_collection
+        .extend([&new_fingerprints[..], &other_flavors_fingerprints[..]].concat());
```

Comment:
@tarkah – This is a place I could use your help for optimizing.
What is happening is that i've extended our type we save to `fingerprints.yml` with flavor. Before, everytime we switched flavor, we removed the fingerprints. Now we have them all, for both flavors.
That means I have to do some filtering on our fingerprint_collection when using it, and when saving `new_fingerprints` i need to save "the others" as well, otherwise we would only have either `retail` or `classic` – however it's not that pretty the solution ive made right now so if you have some idea up your sleeve, please let me know or push a fix.

Reply:
We need to change fingerprint collection into a hashmap with flavor as key as well. This way we can just replace the Vec<Fingerprint> value for that flavor key and save the collection. No need to worry about the other flavor since that side won't get modified.

Reply:
Maybe it's something you can help with? We need to think about mirgation, so the current users then so they don't have to do anything.

Reply:
Yeah I can help. With your change to the Persistent Data trait, it won't fail. New fingerprints will just get generated on first launch, which shouldn't be a big deal for a new release.

Reply:
@tarkah you are the best. I won't touch anything in `parse.rs` until you have pushed then to avoid a nasty merge.
I'll focus on everything around and make sure that is working and plays nicely.

Reply:
See #109